pkg/k8s/api_test.go

```diff
@@ -4,26 +4,86 @@ import (
 	"testing"
 
 	"github.com/runconduit/conduit/pkg/shell"
+	"fmt"
+	"net/url"
 )
 
+type TestFixture struct{
+	testInput string
+	expectedOutput string
+}
 func TestKubernetesApiUrlFor(t *testing.T) {
+	const namespace = "some-namespace"
+	const extraPath = "/some/extra/path"
+	testData := []TestFixture{
+		{
+			testInput: "https://35.184.231.31",
+			expectedOutput:  fmt.Sprintf("https://35.184.231.31/api/v1/namespaces/%s%s", namespace, extraPath),
+		},
+		{
+			testInput: "35.184.231.31",
+			expectedOutput:  fmt.Sprintf("https://35.184.231.31/api/v1/namespaces/%s%s", namespace, extraPath),
+		},
+		{
+			testInput: "http://35.184.231.31",
+			expectedOutput:  fmt.Sprintf("http://35.184.231.31/api/v1/namespaces/%s%s", namespace, extraPath),
+		},
+	}
+
 	t.Run("Returns URL from base URL overridden in construction", func(t *testing.T) {
-		namespace := "some-namespace"
-		extraPath := "/some/extra/path"
-		expectedUrlString := "https://35.184.231.31/api/v1/namespaces/some-namespace/some/extra/path"
+		for _, data := range testData {
+			actualUrl := generateURL(data, t, namespace, extraPath)
+			if actualUrl.String() != data.expectedOutput {
+				t.Fatalf("Expected generated URL to be [%s], but got [%s]", data.expectedOutput, actualUrl.String())
+			}
+		}
+	})
 
-		api, err := NewK8sAPi(shell.NewUnixShell(), "testdata/config.test", "https://35.184.231.31")
-		if err != nil {
-			t.Fatalf("Unexpected error starting proxy: %v", err)
+	t.Run("Returns unmodified URL from base URL overridden in construction", func(t *testing.T){
+		/* NewK8sAPI is agnostic to the scheme used for the override API. It is only contains code that conveniently
+		adds the default scheme used to connect with the k8s API (https). Supplying a URL with a scheme other than https
+		should return an unmodified override URL */
```

Comment:
TIOLI I wonder if this code comment would be made redundant if we rename the test case to something like `"Doesnt modify base URL if it already has a scheme"`?